bin/cli.js

```diff
@@ -1,17 +1,256 @@
+  }
+  if (!config.environments || config.environments.length === 0)
+    config.environments = [ 'node' ];
+};
+
+const getConfig = () => {
+  const version = parseFile(path.resolve(__dirname, '../package.json')).version;
+  program.version(version).usage('[options] -- <file ...>');
+  cliOptions.forEach(option => program.option(...option));
+  program.parse(process.argv);
+
+  const config = program.config ? parseFile(program.config) : {};
+  getEnvironment(config, program);
+
+  config.files = merge(config.files, program.args);
+  config.files = loadFiles(config.files);
+
+  config.exclude = merge(config.exclude, program.exclude);
+  config.files = exclude(config.files, config.exclude);
+  config.nodeOnly = merge(config.nodeOnly, program.nodeOnly);
+  config.browserOnly = merge(config.browserOnly, program.browserOnly);
+
+  if (config.environments.includes('browser')) {
+    config.browser = config.browser || {};
+
+    config.browser.browsers = merge(
+      config.browser.browsers, program.browsers);
+    config.browser.logLevel = program.browserLog || config.browser.logLevel;
+    config.browser.port = +program.browserPort || config.browser.port;
+    config.browser = getBrowserConfig(config);
+  }
+  return config;
+};
+
+const runNode = (config, cb) => {
+  metatests.runner.instance.on('finish', () => {
+    console.log('Tests finished. Waiting for unfinished tests after end...\n');
+    setTimeout(() => cb(metatests.runner.hasFailures ? 1 : 0), 5000);
+  });
+  console.log(`\nNode ${process.version} (v8 ${process.versions.v8}):`);
+  merge(config.files, config.nodeOnly)
+    .map(name => require(path.resolve('./' + name)));
+};
+
+const runBrowser = (config, cb) => {
+  const buildDir = path.resolve('./build');
+  const buildAdapter = path.resolve('./build/adapter.js');
+
+  if (!fs.existsSync(buildDir)) fs.mkdirSync(buildDir);
+  const required = merge(config.files, config.browserOnly)
+    .map(file => `require('../${file}');`)
+    .join('');
+
+  fs.writeFileSync(buildAdapter, '__karma__.start=()=>{};' + required);
+  const server = new karma.Server(config.browser, code => {
+    fs.unlinkSync(buildAdapter);
+    fs.rmdirSync(buildDir);
+    cb(code);
+  });
+  server.start();
+};
+
+const runBoth = (config, cb) => {
+  runBrowser(config, codeB => {
+    console.log('Metatests in browser finished with code', codeB);
+    runNode(config, codeN => {
+      console.log('Metatests in node finished with code', codeN);
+      cb(codeB + codeN);
+    });
+  });
+};
+
+const onExit = code => {
+  console.log('Metatests finished with code', code);
+  process.exit(code);
+};
+
+const config = getConfig();
+console.log('Metatest CLI Stub');
+console.log(`Metatests final config:\n${JSON.stringify(config, null, 2)}\n`);
+if (!config.files.length) {
+  program.outputHelp(help => 'No test files specified\n\n' + help);
+  onExit(1);
+}
+
+if (config.environments.length === 2) runBoth(config, onExit);
```

Comment:
I think this check may be unreliable in some cases, in particular, when a user explicitly provides the environments in the config.

For example, the user can create a config in which environments will be an array `['node', 'something-else']`, and this will still run both `'node'` and `'browser'` tests, which is, in fact, incorrect. Other possible invalid configuration would be providing more than two values in the environments field, for example `['node', 'browser', 'something-else', 'some-other-value']`. There are two issues with such configuration, the first one being the fact that the tests will be run in only one of the environments, and the second one is the fact that the environment to run the test in is going to be determined by the order in which these test environments are presented in the user provided configuration.

We have two options on how to deal with such configurations. The first one is to use only the environments that are currently implemented and warn the user about the environments, which they provided, that are not available. The second option is to completely disallow the use of the non-available environments by refusing to run the tests while there is at least one unknown environment present in the configuration.
I am fine with both of the above solutions to this problem, so it's up to you and others to decide on which behavior is more appropriate for such a situation.